app/controllers/public/orders_controller.rb

```diff
@@ -52,14 +57,22 @@ def confirm
       @order.address = current_customer.address
       @order.address_name = current_customer.first_name + current_customer.last_name
     elsif params[:order][:address_option] == "1"
-      @address = Address.find(params[:customer_address])
-      @order.postal_code = @address.postal_code
-      @order.address = @address.address
-      @order.address_name = @address.address_name
+      if params[:customer_address].present?
+        @address = Address.find(params[:customer_address])
+        @order.postal_code = @address.postal_code
+        @order.address = @address.address
+        @order.address_name = @address.address_name
+      else
+        redirect_to new_order_path, alert: "登録済住所がありません"
+      end
     else
-      @order.postal_code = params[:order][:postal_code]
-      @order.address = params[:order][:address]
-      @order.address_name = params[:order][:address_name]
+      if params[:order][:postal_code].present? && params[:order][:address].present? && params[:order][:address_name].present?
+        @order.postal_code = params[:order][:postal_code]
+        @order.address = params[:order][:address]
+        @order.address_name = params[:order][:address_name]
+      else
+        redirect_to new_order_path, alert: "全て入力してください"
+      end
```

Comment:
バリデーションチェックを最後にするので、ここで値のチェックは不要です
```suggestion
        @order.postal_code = params[:order][:postal_code]
        @order.address = params[:order][:address]
        @order.address_name = params[:order][:address_name]
```